skbio/alignment/_pairwise.py

```diff
@@ -189,8 +88,8 @@ def local_pairwise_align_nucleotide(seq1, seq2, gap_open_penalty=5,
     # use the substitution matrix provided by the user, or compute from
     # match_score and mismatch_score if a substitution matrix was not provided
     if substitution_matrix is None:
-        substitution_matrix = \
-            make_identity_substitution_matrix(match_score, mismatch_score)
+        substitution_matrix = SubstitutionMatrix.identity(
+            'ACGTU', match_score, mismatch_score).to_dict()
```

Comment:
It doesn't need to happen in this PR, but it would be nice if string literals were represented in stable variables either at the object scope or module scope (e.g., this would change to `'ACGTU'` -> `self.NUCLEOTIDES` or something to that effect. The benefit is often these types of strings are reused, and centralizing them facilitates management and reduces chance of bugs from replication/maintenance errors

Reply:
This is a good suggestion. Actually, the original skbio codebase already has it. It is the `definite_chars` attribute of each sequence type. For example see [here](https://github.com/scikit-bio/scikit-bio/blob/8a72c20c6ad50f9cb6d44f5fcf87be5f9582de58/skbio/sequence/_dna.py#L166C29-L166C29). Note: I made the edit here only temporarily, as the entire Python implementation of alignment algorithms will be removed and replaced soon (maybe in the next 2-3 PRs).